### DISCUSSION TITLE
Find shipping cost for Etsy purchased shipping

### DISCUSSION THREAD

**Discussion by ChristopherLuciano** [etsy/open-api#1211] [Q&A]
Hello,

I have ledger entries with ledger_type="shipping_labels" and reference_type="shipping_label", and i believe the reference_id fields there are the label number.

How to associate this reference_id / label number with the order for which it was purchased?  Where to get the actual shipping cost associated with an order?

I'm trying to put together a simple financial utility for myself which shows each order's gross, fees, costs, net.  This would obviously be incomplete without capturing the shipping cost for each order.

Thanks in advance!

**Comment by natesooter** [comment]
+1, I am running into this same question. This also applies to costs such as USPS shipping adjustments, where we get a shipping labels reference type. Need more info to calculate full costs for an order.

**Comment by tdiggity** [comment]
+1

**Comment by NickCis** [comment]
No update regarding this? :/

**Comment by GermanPearls** [comment]
Pinging this. Would also like to be able to link shipping costs to an order. Is there a way yet to do this via export or API?

**Comment by Mikey500** [comment]
Did you find any further guidance on this? Thanks

**Comment by ivordurham** [comment]
I wrote up my solution here in answer to a similar issue: https://github.com/etsy/open-api/discussions/1232#discussioncomment-17872677
